internal/controller/ingress/domain_controller.go

```diff
@@ -137,6 +137,12 @@ func (r *DomainReconciler) create(ctx context.Context, domain *ingressv1alpha1.D
 func (r *DomainReconciler) update(ctx context.Context, domain *ingressv1alpha1.Domain) error {
 	resp, err := r.DomainsClient.Get(ctx, domain.Status.ID)
 	if err != nil {
+		// If the domain is gone, clear the status and trigger a re-reconcile
+		if ngrok.IsNotFound(err) {
+			domain.Status = ingressv1alpha1.DomainStatus{}
+			return r.controller.ReconcileStatus(ctx, domain, err)
```

Comment:
Q. Do we want to do the regular exponential backoff here or should we set our own backoff time such as `3 * time.Minute`?

Reply:
I think the regular exp backoff should be fine. I checked and it caps it at 1000 seconds. All the reconcilers should be clearing the status ID on 404s and then it should go through the create flow. If there is a persistent problem updating the Status on the CR coming from k8s, I'm not sure if there is much we can do.